dfsnew/commit.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015 The Serviced Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dfs
+
+import (
+	"errors"
+
+	"github.com/control-center/serviced/dfsnew/docker"
+	"github.com/zenoss/glog"
+)
+
+var (
+	ErrRunningContainer = errors.New("container is running")
+	ErrStaleContainer   = errors.New("container is stale")
+)
+
+// Commit commits a container spawned from the latest docker registry image
+// and updates the registry.  Returns the affected registry image.
+func (dfs *DistributedFilesystem) Commit(ctrID string) (string, error) {
```

Comment:
:+1:  - this is a great example of easy-to-read code + complete unit-test coverage